File: backend-dotnet/SahayaSetu/Models/Donation.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SahayaSetu.Models
{
    public class Donation
    {
        [Key]
        public int DonationId { get; set; }

        [Required]
        public int DonorId { get; set; }

        [ForeignKey(nameof(DonorId))]
        public Donor Donor { get; set; } = null!;

        [Required]
        public int FundraiserRequestId { get; set; }

        [ForeignKey(nameof(FundraiserRequestId))]
        public FundraiserRequest FundraiserRequest { get; set; } = null!;

        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal Amount { get; set; }

        [Required]
        public PaymentMode PaymentMode { get; set; }

        public string? PaymentReference { get; set; }

        public DateTime DonatedAt { get; set; } = DateTime.UtcNow;

        public bool ResponseStatus { get; set; } = false;

        public string? ResponseMessage { get; set; }

        public DateTime? ResponseSentAt { get; set; }
    }
}

File: backend-java/sahayaSetu/src/main/java/com/sahayaSetu/entities/Donation.java
package com.sahayaSetu.entities;

import com.sahayaSetu.entities.enums.PaymentMode;
import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "donations")
@Data
public class Donation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "donation_id")
    private Long donationId;

    @ManyToOne
    @JoinColumn(name = "donor_id", nullable = false)
    private Donor donor;

    @ManyToOne
    @JoinColumn(name = "fundraiser_request_id", nullable = false)
    private FundraiserRequest fundraiserRequest;

    @Column(nullable = false)
    private BigDecimal amount;

    @Enumerated(EnumType.STRING)
    @Column(name = "payment_mode", nullable = false)
    private PaymentMode paymentMode;

    @Column(name = "payment_reference")
    private String paymentReference;

    @CreationTimestamp
    @Column(name = "donated_at", updatable = false)
    private LocalDateTime donatedAt;

    @Column(name = "response_status")
    private Boolean responseStatus = false;

    @Column(name = "response_message", columnDefinition = "TEXT")
    private String responseMessage;

    @Column(name = "response_sent_at")
    private LocalDateTime responseSentAt;
}
